src/core/struct_tree.js

```diff
@@ -431,67 +413,75 @@ class StructTreeRoot {
         if (Number.isInteger(kid) && updateElement(kid, pageKid, kidRef)) {
           break;
         }
+        if (!(kid instanceof Dict)) {
+          continue;
+        }
+        let type = kid.get("Type");
+        type = type instanceof Name ? type.name : null;
+        if (type !== "MCR") {
+          break;
+        }
```

Comment:
In this case, where we want to check for a *particular* Name, would it make sense to shorten this slightly?
```suggestion
        if (!isName(kid.get("Type"), "MCR")) {
          break;
        }
```